roles/update_etc_hosts/meta/argument_specs.yml

```diff
@@ -0,0 +1,63 @@
+---
+# Requires: ansible 2.11
+# Argument specifications in this separate file maintain backwards compatibility.
+argument_specs:
+
+# For required variables using aliases do not work and fail the argument validation.
+
+  main:
+    short_description: Configure /etc/hosts
+    options:
+      update_etc_hosts_list:
+        type: list
+        description:
+          - List of nodes, ip adresses aliases and comments to define in /etc/hosts
+          - This is required for proper local host configuration.
+        elements: dict
+        options:
+          node_ip:
+            description:
+              - IP address of the node
+          node_name:
+            description:
+              - Hostame of the node
+          node_domain:
+            description:
+              - Domainname of the node
+              - _Optional. Defaults to sap_domain_
+          aliases:
+            type: list
+            description:
+              - List of hostname aliases
+            elements: str
+          alias_mode:
```

Comment:
As per my understanding hostname aliases are not supported by SAP software. 
That is why my recommendation will be to remove this in SAP related collection

Reply:
Host alias are ok, but should not be used for configuring SAP systems